classes/class3/cat-app/app.js

```diff
@@ -0,0 +1,36 @@
+var express = require('express');
+
+var path = require('path');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+var exphbs = require('express-handlebars');
+var mongoose = require('mongoose');
+
+var index = require('./routes/index');
+var cats = require('./routes/cats')
+
+var app = express();
+var mongoURI = process.env.MONGOURI || "mongodb://localhost/cats";
+mongoose.connect(mongoURI);
+
+var PORT = process.env.PORT || 3000;
+
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.get('/', index.home);
+app.get('/cats', cats.list);
+app.get('/cats/new', cats.create);
+app.get('/cats/bycolor/*', cats.colorSort);
```

Comment:
Try to use a named wildcard in the future, for example: `'/cats/bycolor/:color'`
This will allow you to access the value in your colorSort function as `req.params.color`.
It will also prevent something like `'/cats/bycolor/some/extra/paths'` from being treated as a valid path.